plugins/newsletter-from-post/email-settings.tsx

```diff
@@ -0,0 +1,253 @@
+    nb_newsletter_list?: string[];
+    nb_newsletter_header_img?: number;
+  };
+  content?: string;
+}
+
+function EmailSettings() {
+  const [type, setType] = useState('');
+  const [subject, setSubject] = useState('');
+  const [preview, setPreview] = useState('');
+  const [list, setList] = useState<string[]>([]);
+  const [posting, setPosting] = useState(false);
+  const [hasChanges, setHasChanges] = useState(false);
+  const [content, setContent] = useState('');
+  const [image, setImage] = useState(0);
+  const [link, setLink] = useState('');
+
+  const listArray = Array.isArray(list) ? list : [list];
+
+  const [lists, setLists] = useState<ListResult[]>([]); // eslint-disable-line
+
+  const [meta, setMeta] = usePostMeta();
+  const {
+    nb_newsletter_breaking_post_id: breakingPostId,
+  } = meta;
+
+  const { postId } = useSelect((select) => {
+    const { getEditedPostAttribute } = select('core/editor') as CoreEditor;
+    return {
+      postId: getEditedPostAttribute('id'),
+    };
+  }, []);
+
+  const setSelectedLists = ((newValue: Array<Option>) => {
+    const listIds = newValue.map((item: Option) => item.value);
+    setList(listIds);
+    setHasChanges(true);
+  });
+
+  const listsToOptions = (rawLists: ListResult[]) => {
+    const output = rawLists.map((item: ListResult) => ({ label: item.Name, value: item.ListID }));
+    return output;
+  };
+
+  const options = lists.length > 0 ? listsToOptions(lists) : [];
+  const selected = options.filter((item: Option) => listArray.includes(item.value));
+
+  // Should run onload to get the post meta values.
+  useEffect(() => {
+    if (!breakingPostId) {
+      return;
+    }
+    apiFetch(
+      { path: `/wp/v2/nb_newsletter/${breakingPostId}` },
+    ).then((response) => {
+      const {
+        meta: {
+          // @ts-ignore
+          nb_newsletter_email_type: emailType,
+          // @ts-ignore
+          nb_newsletter_subject: emailSubject,
+          // @ts-ignore
+          nb_newsletter_preview: emailPreview,
+          // @ts-ignore
+          nb_newsletter_list: emailList,
+          // @ts-ignore
+          nb_newsletter_header_img: emailImage,
+        } = {},
+        link: newsletterLink = '',
+      } = response as WP_REST_API_Post;
+      setType(emailType);
+      setSubject(emailSubject);
+      setPreview(emailPreview);
+      setList(emailList);
+      setImage(emailImage);
+      setLink(newsletterLink);
+    });
+  }, []); // eslint-disable-line
+
+  useEffect(() => { // eslint-disable-line
+    if (lists.length > 0) {
+      return;
+    }
+    apiFetch({ path: '/newsletter-builder/v1/lists' }).then((response) => {
+      setLists(response as any as ListResult[]);
+    });
+  }, [lists]);
+
+  const updateNewsletter = () => {
+    if (posting) {
+      // setRetry(true);
+      return;
+    }
+    setPosting(true);
+    if (!postId) {
+      return;
+    }
+    if (!subject && !list && !content) {
+      return;
+    }
```

Comment:
Do you need to set posting to false in these cases?
```suggestion
    if (!postId) {
      setPosting(false);
      return;
    }
    if (!subject && !list && !content) {
      setPosting(false);
      return;
    }
```

Reply:
I just moved the `setPosting(true);` to after the returns.